### ISSUE TITLE
loss value  is too high

### ISSUE THREAD

**Issue by VeeranjaneyuluToka** [emadboctorx/yolo-tf2#22]
I am using the yolo4.cfg on my own classes and looks like initial few epochs loss is too high as below and looks like loss is increasing further.
1713/1713 [==============================] - 511s 286ms/step - loss: 11212826311761854464.0000 - layer_373_lambda_loss: 8495524272942594048.0000 - layer_348_lambda_loss: 2233597716050167552.0000 - layer_323_lambda_loss: 483693073789309312.0000 - val_loss: 23797224081195008.0000 - val_layer_373_lambda_loss: 17400711033651200.0000 - val_layer_348_lambda_loss: 6277336295014400.0000 - val_layer_323_lambda_loss: 105817038651392.0000

Epoch 00001: saving model to /home/veeru/processing/tf_git_hubs/yolo_tf/yolo-tf2-master/models/fish_OD_model.tf
Epoch 2/100
1713/1713 [==============================] - 487s 283ms/step - loss: 33454053496164884.0000 - layer_373_lambda_loss: 25993758855185396.0000 - layer_348_lambda_loss: 4133137610872308.5000 - layer_323_lambda_loss: 3313808009742410.0000 - val_loss: 22435384440913920.0000 - val_layer_373_lambda_loss: 20205774575763456.0000 - val_layer_348_lambda_loss: 743358309859328.0000 - val_layer_323_lambda_loss: 1472874745430016.0000

Any idea what could be the reason?

Thanks,
Veeru.

**Comment by emadboctorx** [comment]
What hyperparameters are you using? what are the steps to reproduce the results? what versions? what platform? dataset sample?

**Comment by VeeranjaneyuluToka** [comment]
Hi, Thanks for your quick reply!
Am running the below command which gives some idea of a few hyperparameters.
yolotf2 train --input-shape "(416, 416, 3)" --classes "/data/database-building/Veeru_data/regression_data/train_data/class.names" --model-cfg "yolo_tf2/config/yolo4.cfg" --dataset-name "my_OD" --relative-labels "/home/veeru/processing/tf_git_hubs/yolo_tf/yolo-tf2-master/data/xml_labels/train_validation_merge.csv"  --epochs 100 --batch-size 8 --learning-rate 1e3 --merge-evaluation --min-overlaps 0.5 --test-size 0.2 --augmentation-preset PRESET1 --image-folder /home/veeru/processing/tf_git_hubs/yolo_tf/yolo-tf2-master/data/photos

I just did pip install . as you suggested in readme and upgraded from tf_v2.3 to tf_v2.4 as i am using 3090 nvidia gpu card which needs to have tf_v2.4
As i mentioned above, it is my own data which has 11 classes and 17126 samples after augmentation.
Let me know if you need any more information? 

Thanks,
Veeru.

**Comment by emadboctorx** [comment]
Try several runs: one with yolov3 without augmentation and let me know what the results are. I will also need to check a sample of the dataset you're using and this csv file `train_validation_merge.csv`

**Comment by VeeranjaneyuluToka** [comment]
Looks like something wrong with no.of classes, i have 11 classes but by default it is 80. So if i do not pass 11 some how, it will use 80 classes, is not it?

**Comment by emadboctorx** [comment]
You do not pass anything, the number of classes is inferred during runtime using `/data/database-building/Veeru_data/regression_data/train_data/class.names` which I will also need to check.

**Comment by VeeranjaneyuluToka** [comment]
I think you are right, it is reading from the class file and passing the len of that list to base class here in the below code
 self.class_names = [
            item.strip() for item in open(self.classes_file).readlines()
        ]
        super().__init__(
            input_shape,
            model_configuration,
            **len(self.class_names)**,
            anchors,
            masks,
            max_boxes,
            iou_threshold,
            score_threshold,
        )

**Comment by emadboctorx** [comment]
> Try with yolov3 without augmentation and let me know what the results are. I will also need to check a sample of the dataset you're using and this csv file `train_validation_merge.csv`

+ the classes file please

**Comment by VeeranjaneyuluToka** [comment]
Am not able to attach the file as such, but you can create a simple file with name as mentioned in the command with below classes.
road
building
tree
water
cable
bridge
misc
ground
scenery
machine
sunlight

**Comment by emadboctorx** [comment]
I'm closing this issue as it looks like it has staled.
